company-box.el

```diff
@@ -784,7 +784,8 @@ It doesn't nothing if a font icon is used."
 (defun company-box-show (&optional on-update)
   (unless on-update
     (setq company-box--parent-start (window-start))
-    (add-hook 'window-scroll-functions 'company-box--handle-scroll-parent nil t))
+    (add-hook 'window-scroll-functions 'company-box--handle-scroll-parent nil t)
+    (set-frame-width (company-box--get-frame) company-tooltip-minimum-width))
```

Comment:
@leodag Sorry I wasn't clear. My question is about this line of code:
```el
(set-frame-width (company-box--get-frame) company-tooltip-minimum-width))
```
Why do we need this `set-frame-width` ? 
`company-box` does respect `company-tooltip-minimum-width` without this line

Reply:
So that, when the tooltip appears, it will have the minimum possible width; and grow as you scroll down, but never shrink until it is invoked again. Otherwise it will never shrink, even if the tooltip was hidden and shown again.

Here's an example:
1. I initiate a completion. company-box has width 16 (my minimum)
2. I scroll a wider item into view, making company-box have width 20
3. I cancel the completion with C-g. Nothing is completed
4. I initiate a completion. company-box has width x

At this point, I want x to be 16 (company itself does it like that). Without this set-frame-width, x will be 20.

Reply:
@leodag Ok I understand now thanks.
Though I do not observe the behavior you describe.
I followed all the steps you describe, and when I re-initiate a completion after cancelling one, the width is the smallest possible one (width of the biggest completion item visible), it doesn't keep the width of the canceled frame:

![Peek 2022-09-10 10-51](https://user-images.githubusercontent.com/5273820/189476287-082c474c-ada9-4465-b7e9-9ab8c83778a5.gif)

Here I cancel the completion at `input`, and when I re-initiate a completion with `inputs`, the width is smaller.

Could we keep in this PR only the changes for `company-tooltip-width-grow-only`, and open a new issue or PR for the issue you are describing ?